File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int N = height.size(); int a = 0; int b = N - 1; int res = 0;

        while (a < b) {
            int area = min(height[a], height[b]) * (b - a);
            res = max(area, res);

            if (height[a] < height[b]) {
                a ++;
            } else {
                b --;
            }
        }

        return (res);
    }
};

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        a, b = 0, len(height) - 1
        res = 0
        while b > a:
            res = max(res, min(height[a], height[b]) * (b - a))

            if height[a] > height[b]: b -= 1
            else: a += 1
        
        return res